### DISCUSSION TITLE
link-external-icon and link-external-newwindow feature broken in versions after "quarto-1.4.330"?

### DISCUSSION THREAD

**Discussion by drganghe** [quarto-dev/quarto-cli#6980] [Q&A] [answered]
### Description

I rerendered my site using the most recent developed version and found all link-external-icon and link-external-newwindow features are gone.

I tried to specify for the whole site in _quarto.yml or specific pages, neither works.

```
format:
  html:
    link-external-icon: true
    link-external-newwindow: true
```

If downgrade to the stable released version, both features are back.

**Comment by dragonstyle** [comment]
Thank you for letting us know- this should be fixed in our next pre-release build!

**Reply by drganghe** [reply]
Thank you. It fixed the link-external-icon and link-external-newwindow behavior. However, the right nav bar and right footer nav are messed up. Those should just show the icons, no links.

**Reply by dragonstyle** [reply, marked as answer]
That is a regression in our pre-release, thank you for catching it! We are building a fresh pre-release which will not show this problem (and which includes the proper behavior for external links). Thanks again!

**Reply by drganghe** [reply]
This is great, thank you!

**Reply by drganghe** [reply]
Just one more small feedback. Email addresses are currently identified as external link therefore added a external icon, which should be excluded.

**Reply by dragonstyle** [reply]
Thanks that will be corrected in our next pre-release build...

**Reply by cod3licious** [reply]
I think this bug is back - at least for `render` - `preview` works fine.

In my `_quarto.yml` I have
```
format:
  html:
    link-external-newwindow: true
    link-external-icon: true
```
which results in the expected icons and opening behavior when I look at a page with `quarto preview` but the icons are gone when opening the resulting html that is created with `quarto render` in a browser and there the links are also opened in the same window.

I'm using quarto version `1.6.39` on macOS 14.6.1 (M1 Pro).

**Reply by cscheid** [reply]
If you're using `quarto render` then you also need to make sure that you're looking at the result being served from a web server. Are you double-clicking the .html file on Finder instead, by chance?

**Reply by cderv** [reply]
@cod3licious this is an old discussion. Can you open a new issue with a reproducible example or step to reproduce ? 

Thank you !

**Reply by cod3licious** [reply]
@cscheid oh, okay, sorry, yeah I only manually opened the html file in Chrome - I didn't think something as simple as appending `target="_blank"` to a link would mean that it only works when being served from a web server 🙈

**Reply by cscheid** [reply]
It _might_ be the case that we have a regression. But we don't support the `file://` protocol in general, so you need to check with (for example), running `python -m http.server` on the directory where the files are built, and then navigate to (for example) `http://127.0.0.1:8000`

**Reply by cod3licious** [reply]
When accessing the page through `python -m http.server` it works 🙌. Thank you and sorry for the false alarm 🙏. Still figuring out the quirks of quarto after years of asciidoctor 🫣

**Reply by cscheid** [reply]
This isn't a Quarto quirk; this is simply your web browser interpreting HTML, and any other tools would behave the same.

**Reply by cod3licious** [reply]
I guess the HTML pages generated by asciidoctor were more self-contained in the sense that there wasn't a big difference when opening the HTML in the browser locally vs. when it was served by a web server, which made it a bit easier to understand what you're getting.
